monitoring-dashboard/components/org.wso2.ei.dashboard.core/src/main/java/org/wso2/ei/dashboard/micro/integrator/delegates/LogsDelegate.java

```diff
@@ -23,40 +23,93 @@
 import org.wso2.ei.dashboard.core.rest.model.LogListInner;
+import org.wso2.ei.dashboard.core.rest.model.LogsResourceResponse;
+import org.wso2.ei.dashboard.micro.integrator.commons.DelegatesUtil;
 import org.wso2.ei.dashboard.micro.integrator.commons.Utils;
 
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
 import java.util.List;
 import java.util.concurrent.atomic.AtomicBoolean;
 
 /**
  * Delegate class to handle requests from log files page.
  */
 public class LogsDelegate {
-    private static final Log log = LogFactory.getLog(LogsDelegate.class);
-    private final DatabaseManager databaseManager = DatabaseManagerFactory.getDbManager();
+    private static final Log logger = LogFactory.getLog(LogsDelegate.class);
+    private static final DataManager dataManager = DataManagerSingleton.getDataManager();
+    private static List<LogListInner>  searchedList;
+    private static String prevSearchKey = null;
+    private static int count;
+
+    public LogsResourceResponse getPaginatedLogsListResponse(String groupId, List<String> nodeList, String searchKey, 
+        String lowerLimit, String upperLimit, String order, String orderBy, String isUpdate) 
+        throws ManagementApiException {
+        
+        logger.debug("Fetching Searched Endpoints from MI.");
+        logger.info("group id :" + groupId + ", lowerlimit :" + lowerLimit + ", upperlimit: " + upperLimit);
+        logger.info("Order:" + order + ", OrderBy:" + orderBy + ", isUpdate:" + isUpdate);
+        int fromIndex = Integer.parseInt(lowerLimit);
+        int toIndex = Integer.parseInt(upperLimit);
+        boolean isUpdatedContent = Boolean.parseBoolean(isUpdate);
+
+        logger.debug("prevSearch key :" + prevSearchKey + ", currentSearch key:" + searchKey);
+
+        if (isUpdatedContent || prevSearchKey == null || !(prevSearchKey.equals(searchKey))) {
+            searchedList = getSearchedLogsListFromMI(groupId, nodeList, searchKey, order, orderBy);
+            count = getLogsCount(searchedList);
+        }
+        LogsResourceResponse logsResourceResponse = new LogsResourceResponse();
+        LogList paginatedList = getPaginationResults(searchedList, fromIndex, toIndex);
+        logsResourceResponse.setResourceList(paginatedList);
+        logsResourceResponse.setCount(count);
+        prevSearchKey = searchKey;
+        return logsResourceResponse;
+    }
+
+    private int getLogsCount(List<LogListInner> loglist) {
+        int count = 0;
+        List<String> logsNames = new ArrayList<>();
+        String name;
+        for (LogListInner log : loglist) {
+            name = log.getName();
+            if (!logsNames.contains(name)) {
+                count++;
+                logsNames.add(name);
+            }
+        }
+        return count;
+    }
+    public static List<LogListInner> getSearchedLogsListFromMI(String groupId, List<String> nodeList,
```

Comment:
Let's keep a new line between the two methods

Reply:
done